viamonvif/discovery.go

```diff
@@ -274,31 +282,47 @@ func GetCameraInfo(
 	return cameraInfo, nil
 }
 
-// GetRTSPStreamURIsFromProfiles uses the ONVIF Media service to get the RTSP stream URLs for all available profiles.
-func GetRTSPStreamURIsFromProfiles(
+// GetRTSPStreamInfoFromProfiles uses the ONVIF Media service to get the RTSP stream URLs for all available profiles.
+func GetRTSPStreamInfoFromProfiles(
 	ctx context.Context,
 	dev OnvifDevice,
 	creds device.Credentials,
 	logger logging.Logger,
-) ([]string, error) {
+) ([]URI, error) {
 	resp, err := dev.GetProfiles(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	// Resultant slice of RTSP URIs
-	var rtspUris []string
-
+	uris := make([]URI, 0)
 	// Iterate over all profiles and get the RTSP stream URI for each one
 	for _, profile := range resp.Profiles {
-		uri, err := dev.GetStreamURI(ctx, profile, creds)
+		streamURI, err := dev.GetStreamURI(ctx, profile.Token, creds)
 		if err != nil {
 			logger.Warn(err.Error())
 			continue
 		}
 
-		rtspUris = append(rtspUris, uri.String())
+		snapshotURI, err := dev.GetSnapshotURI(ctx, profile.Token, creds)
```

Comment:
TODO(seanp): If snapshot lookup fails, we still want to save streamURI and return config.